svc-device-manager/rest/postSimpleUpdate.go

```diff
@@ -0,0 +1,100 @@
+package rest
+
+import (
+	"devicemanager/config"
+	"devicemanager/rest/redfish"
+	"encoding/json"
+	"fmt"
+	"github.com/kataras/iris/v12"
+	"github.com/kataras/iris/v12/context"
+	"github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+)
+
+type postSimpleUpdateHandler struct {
+	cfg config.Config
+}
+
+func (r *postSimpleUpdateHandler) handle(ctx iris.Context) {
+	if ctx.GetHeader("Authorization") == "" {
+		noValidAuthError := "No valid authorization"
+		ctx.StatusCode(http.StatusUnauthorized)
+		logrus.Error(noValidAuthError)
+		ctx.WriteString(noValidAuthError)
+		return
+	}
+
+	var reqInfo redfish.RequestInformation
+
+	// Get information from the request.
+	err := ctx.ReadJSON(&reqInfo)
+	if err != nil {
+		missingInfoError := "Unable to retrieve information from a request: " + err.Error()
+		logrus.Error(missingInfoError)
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.WriteString(missingInfoError)
+		return
+	}
+
+	// Validate Simple Update request
+	err = validateSimpleUpdateAction(reqInfo.Body)
+	if err != nil {
+		simpleUpdateRequestError := "invalid simple update request: %s" + err.Error()
+		logrus.Error(simpleUpdateRequestError)
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.WriteString(simpleUpdateRequestError)
+		return
+	}
+
+	httpClient := redfish.NewHttpClient(r.cfg).WithBasicAuth(reqInfo.Username, string(reqInfo.Password))
+	uri := fmt.Sprintf("https://%s%s", reqInfo.Host, ctx.Request().RequestURI)
+
+	response, err := httpClient.Post(uri, reqInfo.Body)
+	if err != nil {
+		errorMessage := fmt.Sprintf("POST action failed on %s due to: %s", uri, err.Error())
+		logrus.Error(errorMessage)
+		if response == nil {
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(errorMessage)
+			return
+		}
+	}
+
+	defer response.Body.Close()
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		errorMessage := "Error while reading response body: " + err.Error()
+		logrus.Error(errorMessage)
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString(errorMessage)
+		return
+	}
+
+	ctx.StatusCode(response.StatusCode)
+	ctx.Write(body)
```

Comment:
According to the "Redfish Firmware Update White Paper" 6.1 paragraph, service can respond with `202 Accepted` which means that a task is returned in `Location` header. Please forward this task URI in header.